File: src/Entity/Rdv.php
<?php

namespace App\Entity;

use App\Repository\RdvRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;
use Symfony\Component\Validator\Constraints as Assert;
#[ORM\Entity(repositoryClass: RdvRepository::class)]
class Rdv
{
   
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
   
    private ?int $idrdv = null;
 /**
      *  @Assert\NotBlank(message="date de reservation de patient est obligatoire")
     * @Groups("rdv")
     */
   
    #[ORM\Column(type: Types::DATETIME_MUTABLE)]
    private ?\DateTimeInterface $date = null;
   
     /**
      *  @Assert\NotBlank(message="Nom de patient est obligatoire")
     * @Assert\NotBlank(message="NB: nom patient  doit etre Obligatoire et min caractere 7 et max 99")
     * @Assert\Length(
     *      min = 7,
     *      max = 100,
     *      minMessage = "doit etre >=4 ",
     *      maxMessage = "doit etre <=100" )
     * @Groups("rdv")
     */
    
    #[ORM\Column(length: 255)]
    private ?string $nompatient = null;
    /** 
    *@Groups("rdv")
    */

    #[ORM\Column(length: 255)]
    private ?string $numtel = null;
    
     /**
      *  @Assert\NotBlank(message="message de patient est obligatoire")
     * @Assert\NotBlank(message="NB: nom patient  doit etre Obligatoire et min caractere 2 et max 99")
     * @Assert\Length(
     *      min = 2,
     *      max = 100,
     *      minMessage = "doit etre >=4 ",
     *      maxMessage = "doit etre <=100" )
     *   *@Groups("rdv")
     */

    #[ORM\Column(type: Types::TEXT)]
    private ?string $message = null;
    
     /**
      *  @Assert\NotBlank(message="email de patient est obligatoire")
     * @Assert\NotBlank(message="NB: nom patient  doit etre Obligatoire et min  caractere 5 et max 99")
     * @Assert\Length(
     *      min = 5,
     *      max = 100,
     *      minMessage = "doit etre >=4 ",
     *      maxMessage = "doit etre <=100" )
     *   *@Groups("rdv")
     */

    #[ORM\Column(type: Types::TEXT)]
    private ?string $adr = null;



   


   

    public function getIdrdv(): ?int
    {
        return $this->idrdv;
    }

    

    

    public function getDate(): ?\DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(\DateTimeInterface $date): self
    {
        $this->date = $date;

        return $this;
    }

    public function getNompatient(): ?string
    {
        return $this->nompatient;
    }

    public function setNompatient(string $nompatient): self
    {
        $this->nompatient = $nompatient;

        return $this;
    }


    public function getNumtel(): ?string
    {
        return $this->numtel;
    }

    public function setNumtel(string $numtel): self
    {
        $this->numtel = $numtel;

        return $this;
    }

    public function getMessage(): ?string
    {
        return $this->message;
    }

    public function setMessage(string $message): self
    {
        $this->message = $message;

        return $this;
    }

    public function getAdr(): ?string
    {
        return $this->adr;
    }

    public function setAdr(string $adr): self
    {
        $this->adr = $adr;

        return $this;
    }

   

}

File: entities/Rdv.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package edu.connexion3a41.entities;

import java.sql.Date;

/**
 *
 * @author tha3leb
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */


/**
 *
 * @author tha3leb
 */
public class Rdv {
    
     private int idrdv;
    private String nompatient;
     private Date date;
     
    private String numtel;
    private String message;
           
           private String adr;
                private int medecin_id;

    public Rdv(int idrdv) {
        this.idrdv = idrdv;
    }

    public Rdv(int idrdv, String nompatient, Date date, String numtel, String message, String adr, int medecin_id) {
        this.idrdv = idrdv;
        this.nompatient = nompatient;
        this.date = date;
        this.numtel = numtel;
        this.message = message;
        this.adr = adr;
        this.medecin_id = medecin_id;
    }

    public Rdv(String nompatient, Date date, String numtel, String message, String adr, int medecin_id) {
        this.nompatient = nompatient;
        this.date = date;
        this.numtel = numtel;
        this.message = message;
        this.adr = adr;
        this.medecin_id = medecin_id;
    }

   
     public Rdv() {
       
    }

   

   

    public int getIdrdv() {
        return idrdv;
    }

    public void setIdrdv(int idrdv) {
        this.idrdv = idrdv;
    }

    public String getNompatient() {
        return nompatient;
    }

    public void setNompatient(String nompatient) {
        this.nompatient = nompatient;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public String getNumtel() {
        return numtel;
    }

    public void setNumtel(String numtel) {
        this.numtel = numtel;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getAdr() {
        return adr;
    }

    public void setAdr(String adr) {
        this.adr = adr;
    }

    public int getMedecin_id() {
        return medecin_id;
    }

    public void setMedecin_id(int medecin_id) {
        this.medecin_id = medecin_id;
    }

    @Override
    public String toString() {
        return "Rdv{" + "idrdv=" + idrdv + ", nompatient=" + nompatient + ", date=" + date + ", numtel=" + numtel + ", message=" + message + ", adr=" + adr + ", medecin_id=" + medecin_id + '}';
    }

    
           
            
    
}

